test/integ/quote-relay.test.ts

```diff
@@ -0,0 +1,193 @@
+          expect(parseInt(order.info.input.amount.toString())).to.be.lessThan(110000000);
+
+          const { tokenInBefore, tokenInAfter, tokenOutBefore, tokenOutAfter } = await baseTest.executeRelaySwap(
+            alice,
+            filler,
+            order,
+            USDC_MAINNET, // tokenIn
+            USDC_MAINNET, // gasToken
+            USDT_MAINNET // outputToken
+          );
+
+          const expectedMinAmountOut = CurrencyAmount.fromRawAmount(
+            USDT_MAINNET,
+            (quote as RelayQuoteDataJSON).classicQuoteData.quote
+          )
+            .multiply(95)
+            .divide(100); // apply slippage
+
+          const expectedMaxAmountIn = CurrencyAmount.fromRawAmount(
+            USDC_MAINNET,
+            parseInt(order.info.fee.endAmount.toString()) + parseInt(order.info.input.amount.toString())
+          );
+          // at most netMaxAmountIn of tokenIn should be spent
+          expect(
+            tokenInBefore.subtract(tokenInAfter).lessThan(expectedMaxAmountIn) ||
+              tokenInBefore.subtract(tokenInAfter).equalTo(expectedMaxAmountIn)
+          ).to.be.true;
+          // user should have received at least expectedAmountOut of tokenOut
+          expect(
+            tokenOutAfter.subtract(tokenOutBefore).greaterThan(expectedMinAmountOut) ||
+              tokenOutAfter.subtract(tokenOutBefore).equalTo(expectedMinAmountOut)
+          ).to.be.true;
+        });
+
+        it(`stable -> stable, gas token != input token`, async () => {
+          const quoteReq: QuoteRequestBodyJSON = {
+            requestId: 'id',
+            tokenIn: USDC_MAINNET.address,
+            tokenInChainId: 1,
+            tokenOut: USDT_MAINNET.address,
+            tokenOutChainId: 1,
+            amount: await getAmount(1, type, 'USDC', 'USDT', '100'),
+            type,
+            slippageTolerance: SLIPPAGE,
+            configs: [
+              {
+                routingType: RoutingType.RELAY,
+                protocols: ['V2', 'V3', 'MIXED'],
+                swapper: alice.address,
+                gasToken: DAI_MAINNET.address,
+              },
+            ] as RoutingConfigJSON[],
+          };
+
+          const response: AxiosResponse<QuoteResponseJSON> = await call(quoteReq);
+          const {
+            data: { quote },
+            status,
+          } = response;
+
+          const order = new RelayOrder((quote as any).orderInfo, 1);
+          expect(status).to.equal(200);
+
+          order.info.reactor = reactorAddress;
+
+          expect(order.info.swapper).to.equal(alice.address);
+          expect(order.info.input).to.not.be.undefined;
+          expect(order.info.fee).to.not.be.undefined;
+          expect(order.info.universalRouterCalldata).to.not.be.undefined;
+
+          expect(parseInt(order.info.input.amount.toString())).to.be.greaterThan(90000000);
+          expect(parseInt(order.info.input.amount.toString())).to.be.lessThan(110000000);
```

Comment:
nit: duplicate code, this can be wrapped in a helper

Reply:
Going to take in a follow up PR #387